lib/bounces.js

```diff
@@ -47,6 +47,19 @@ module.exports = function (log, error) {
         // But we can measure it!
         log.increment('account.email_bounced.already_verified')
       }
+      // Because we have no session token on this endpoint, metrics context
+      // metadata was stashed against a synthesized token for the benefit of
+      // the activity events. This fake request object allows the correct
+      // metadata to be gathered when we emit the events.
+      var fakeRequestObject = {
```

Comment:
One approach, which I'm not sure how well it would work out in practice, could be as follows.  We know that if the email bounched, there's no way that the `account.confirmed`, `account.signed` etc events can have taken place.  So we look for the data that was stashed for one of those events, clear it (because they're now never going to happen) and log it in the event here.

Hmmm...having typed that out I'm not sure I like it, perhaps we should take a step back and think end-to-end about managing this metrics data, with the new requirements in mind.